helpers/persistPostVisitActions.ts

```diff
@@ -335,4 +437,15 @@ export const persistPostVisitActions = async (
       }
     }
   }
+
+  const unableToEnterData =
+    processData === undefined
+      ? {}
+      : (processData.unableToEnter as Partial<
+          ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]
+        >);
```

Comment:
```suggestion
  const unableToEnterData: Partial<
          ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]
        > =
    processData === undefined
      ? {}
      : processData.unableToEnter;
```

Does this work? If so, let's use it because it's asserting a type rather than casting one.

Reply:
Unfortunately not, as it still thinks `processData.unableToEnter` could be `undefined` (even after adding a more explicit guard for this)

Reply:
```suggestion
  const unableToEnterData: Partial<
          ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]
        > =
    processData === undefined || processData.unableToEnter === undefined 
      ? {}
      : processData.unableToEnter;
```

?

I'm trying to avoid casting as much as possible because it hides errors.

Reply:
I'm getting `Type '_DeepPartialArray<string> | undefined' is not assignable to type 'string[]'.`

Reply:
Ah right. Nevermind then.

Reply:
Should probably be checking that `processData.unableToEnter !== undefined` anyway, though.